doc/external-api/README.md

```diff
@@ -0,0 +1,74 @@
+# Getting started
+
+There are two ways to interface with nix: embedding it, or as a plugin. Embedding means you link one of the nix libraries in your program and use it from there, while being a plugin means you make a library that gets loaded by the nix evaluator, specified through a configuration option.
+
+# Embedding the Nix Evaluator
+
+These examples don't include error handling.
+See the [Handling errors](@ref errors) section for more information.
+
+**main.c:**
+```C
+#include <nix_api_util.h>
+#include <nix_api_expr.h>
+#include <nix_api_value.h>
+#include <stdio.h>
+
+int main() {
+   nix_libexpr_init(NULL);
+
+   Store* store = nix_store_open(NULL, "dummy://", NULL);
+   State* state = nix_state_create(NULL, NULL, store); // empty nix path
+   Value *value = nix_alloc_value(NULL, state);
+
+   nix_expr_eval_from_string(NULL, state, "builtins.nixVersion", ".", value);
+   nix_value_force(NULL, state, value);
+   printf("nix version: %s\n", nix_get_string(NULL, value));
+
+   nix_gc_decref(NULL, value);
+   nix_state_free(state);
+   nix_store_unref(store);
+   return 0;
+}
+```
+ 
+**Usage:**
+```
+$ gcc main.c $(pkg-config nix-expr-c --libs --cflags) -o main
+$ ./main
+nix version 1.2.3
+```
+
+
+# Writing a Nix Plugin
+
```

Comment:
```suggestion
In this example we add a custom primitive operation (*primop*) to `builtins`.
It will increment the argument if it is an integer and return `null` otherwise.

```